### PR TITLE
Convert .md files to .toml files

### LINKED ISSUE #30: Consider using .toml files instead of .md files
a407f821da484bdef0d361770b0c66590935a61a split `features.toml` into many small files ... which of course makes sense maintenance-wise. The data is now however stored in `.md` files with a TOML frontmatter like

```
+++
title = "`debug_assert_ne!`"
rfc_id = 1653
impl_pr_id = 35074
tracking_issue_id = 35073
+++
```

this makes the data harder to work with. If you want to use the data for any other purpose you need to write your own parser for this non-standardized format. It also makes contributing a bit harder ... because it's not immediately clear what format these files have. `.toml` files are well understood by Rust programmers so I think they really are the natural choice.

### PR DESCRIPTION
**Pull request by not-my-profile**
a407f821da484bdef0d361770b0c66590935a61a split features.toml into many
.md files with TOML frontmatters. This commit converts all .md files in
the data/ directory to .toml files so that they can be more easily
processed by other software.

The files were converted with the following Python code:

```python
#!/usr/bin/env python3
import glob

for path in glob.glob('data/**/*.md'):
    text = ''
    with open(path) as f:
        assert next(f) == '+++\n'
        line = next(f)
        while line != '+++\n':
            text += line
            line = next(f)
    with open(path.replace('.md', '.toml'), 'w') as f:
        f.write(text)
```

Closes #30.

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -192,43 +192,18 @@ fn collect_features(
             continue;
         }
 
-        let slug = match file_name.strip_suffix(".md") {
+        let slug = match file_name.strip_suffix(".toml") {
             Some(basename) => basename.to_owned(),
             None => {
-                panic!("expected only .md files and version.toml in data/*, found `{}`", file_name,)
+                panic!(
+                    "expected only .toml files and version.toml in data/*, found `{}`",
```
Comment (jplatte) [inline build.rs:199]:
```suggestion
                    "expected only .toml files in data/*, found `{}`",
```

File: build.rs [hunk 2]
```diff
@@ -192,43 +192,18 @@ fn collect_features(
...
-            }
-        }
-
-        // TODO: Read file contents after frontmatter
 
-        let feature = toml::from_str(&feature_file_frontmatter).with_context(|| {
-            format!("deserializing frontmatter of data/{}/{}", dir_name, file_name)
-        })?;
+        let feature = toml::from_str(&fs::read_to_string(file.path())?)
+            .with_context(|| format!("deserializing of data/{}/{}", dir_name, file_name))?;
```
Comment (jplatte) [inline build.rs:206]:
```suggestion
            .with_context(|| format!("deserializing data/{}/{}", dir_name, file_name))?;
```

**Comment by jplatte** [comment]
Thanks!

**Comment by not-my-profile** [comment]
You're welcome :) Huh, your squash lost my commit message.

**Comment by jplatte** [comment]
Oh yeah, that was not intentional. Shouldn't really be a big deal though.

**Comment by not-my-profile** [comment]
I'm quite the stickler for commit messages^^ But yeah I guess it's fine. Out of curiosity does GitHub just do that by default?

**Comment by not-my-profile** [comment]
Oh btw my commit broke your example links in #16.

**Comment by jplatte** [comment]
> I'm quite the stickler for commit messages^^

Me too but not enough to rewrite history in this case.

> But yeah I guess it's fine. Out of curiosity does GitHub just do that by default?

Not usually.. I don't know what went wrong in this case.

**Pull request merged**